src/status_im/ui/screens/desktop/main/chat/views.cljs

```diff
@@ -148,15 +148,27 @@
            [react/view {:style (message.style/delivery-status outgoing)}
             [message/message-delivery-status message]]])}))))
 
+(def load-step 5)
+
+(defn load-more [messages messages-to-load]
+  (let [all-messages-count (count messages)
+        next-count (min all-messages-count (+ @messages-to-load load-step))]
+    (reset! messages-to-load next-count)))
+
 (views/defview messages-view [{:keys [chat-id group-chat]}]
   (views/letsubs [messages [:get-current-chat-messages-stream]
-                  current-public-key [:get-current-public-key]]
-    (let [chat-id* (atom nil)
-          scroll-ref (atom nil)
+                  current-public-key [:get-current-public-key]
+                  messages-to-load (reagent/atom 5)
+                  chat-id* (reagent/atom nil)]
+    {:component-did-update #(load-more messages messages-to-load)
+     :component-did-mount  #(load-more messages messages-to-load)}
+    (let [scroll-ref (atom nil)
           scroll-timer (atom nil)
           scroll-height (atom nil)
           _ (when (or (not @chat-id*) (not= @chat-id* chat-id))
-              (reset! chat-id* chat-id))]
+              (do
```

Comment:
why is this here in the first place ?

Reply:
Will putting it on a second place  make any difference?

Reply:
no I am asking for the purpose of this code ? Normally you shouldn't have any logic in the view description. So I'm wondering why it has to be there. I think the messages-to-load can be updated in :component-did-update ? What is chat-id* for ? If that is because you can switch chats and the component will still be mounted, I think that means the component logic and parameters are flawed.

Reply:
Yes, this is because changing chat-id means that we switched to new chat and that totally new messages will be loaded. So we need to reset `messages-to-load` counter and again load messages gradually.

Reply:
but if chat-id changes, isn't the component updated again ? at least this logic should also be in component-update ?

Reply:
The problem here is that `:component-did-update` calls after component rendered.
In case when chat-id changed, we need `messages-to-load` value to be updated before first render or all optimization will fail. And looking at react component lifecyle seems that `render()` is appropriate place for this.

Reply:
Per discussion with @yenda, this problem moved to separate issue to be fixed later - #6033